### PR TITLE
Handle proto deps that were added to the Dart SDK

### PR DESCRIPTION
**Pull request by derekxu16**
The Dart SDK will be adding third-party dependencies that it needs to serialize protos in https://dart-review.googlesource.com/c/sdk/+/278200.

### CODE REVIEW

**Comment by derekxu16** [comment]
Here is my planned process for integrating these changes:

1. Get this PR and https://github.com/flutter/buildroot/pull/670 reviewed.
2. Land https://dart-review.googlesource.com/c/sdk/+/278200
3. Once the Dart SDK starts failing to roll into the engine, merge this and https://github.com/flutter/buildroot/pull/670

Plese let me know if this sounds good.

**Comment by zanderso** [comment]
To avoid breaking the Engine tree, the buildroot roll, this engine PR, and a manual Dart roll should be combined into a single PR.

File: DEPS [hunk 1]
```diff
@@ -236,7 +239,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'abada33190daa7914938ec56d59032012fcf12f7',
+  'src': 'https://github.com/flutter/buildroot.git' + '@' + '6d0e54005fac0e5af1a0bbe2426ef49fcb94be28',
```
Comment (zanderso) [inline DEPS:242]: This should be updated to flutter/buildroot tip-of-tree, which is currently https://github.com/flutter/buildroot/commit/c77cac9502fb108be6a1ed4e0e4e1c65649b3a35

**Review by zanderso — APPROVED** [review]
lgtm

**Pull request merged**